codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/CodegenUtils.java

```diff
@@ -145,6 +146,22 @@ public static String getSyntheticTypeNamespace() {
         return CodegenUtils.SYNTHETIC_NAMESPACE;
     }
 
+    /**
+     * Get if the passed in shape is decorated as a synthetic clone, but there is no other shape the clone is
+     * created from.
+     *
+     * @param shape the shape to check if its a stubbed synthetic clone without an archetype.
+     * @return if the shape is synthetic clone, but not based on a specific shape.
+     */
+    public static boolean isStubSyntheticClone(Shape shape) {
+        Optional<SyntheticClone> optional = shape.getTrait(SyntheticClone.class);
+        if (!optional.isPresent()) {
+            return false;
+        }
+
+        SyntheticClone synthClone = optional.get();
+        return synthClone.getArchetype() == null;
```

Comment:
Minor, but we probably want to circle back on this later and have this getter return this as a Optional<T>